x-pack/plugins/security_solution/public/detection_engine/rule_management/logic/use_rule_with_fallback.ts

```diff
@@ -114,11 +116,50 @@ export const useRuleWithFallback = (ruleId: string): UseRuleWithFallback => {
   };
 };
 
+/**
+ * In 8.10.x investigation_fields is mapped as alert, moving forward, it will be mapped
+ * as an object. This util is being used for the use case where a rule is deleted and the
+ * hook falls back to using the alert document to retrieve rule information. In this scenario
+ * we are going to return undefined if field is in legacy format to avoid any possible complexity
+ * in the UI for such flows. See PR 169061
+ * @param investigationFields InvestigationFieldsCombined | undefined
+ * @returns InvestigationFields | undefined
+ */
+export const migrateLegacyInvestigationFields = (
```

Comment:
nit: There are 3 different functions called `migrateLegacyInvestigationFields`, can we differentiate them with more descriptive names?